converter/lookml/dimension.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Optional, Tuple
+
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+from dbt_semantic_interfaces.implementations.elements.dimension import (
+    PydanticDimensionTypeParams,
+)
+from dbt_semantic_interfaces.type_enums.dimension_type import DimensionType
+from dbt_semantic_interfaces.implementations.elements.dimension import (
+    PydanticDimension,
+)
+from pydantic import BaseModel
+
+DEFAULT_TIME_GRANULARITY = TimeGranularity.DAY
+
+
+class LkmlDimension(BaseModel):
+    name: str
+    type: Optional[str]
+
+
+def to_sl_dimensions(dimensions: List[LkmlDimension]) -> List[PydanticDimension]:
+    sl_dimensions: List[PydanticDimension] = []
+    for dimension in dimensions:
+        if not dimension.type:
+            continue
+        dim_type, dim_type_param = get_dimension_type(dimension)
+        sl_dimensions.append(
+            PydanticDimension(
+                name=dimension.name,
+                type=dim_type,
+                type_params=dim_type_param,
+            )
+        )
+    return sl_dimensions
+
+
+def get_dimension_type(
+    lkmlDimension: LkmlDimension,
+) -> Tuple[DimensionType, Optional[PydanticDimensionTypeParams]]:
+    def get_grain(lkmlDimension: LkmlDimension) -> TimeGranularity:
+        if any([t in lkmlDimension.type for t in ["second", "hour", "day"]]):
+            return TimeGranularity.DAY
+        if "week" in lkmlDimension.type:
```

Comment:
nit: can we use the enum value instead of strings (except for second and hour)? e.g., `TimeGranularity.WEEK.value` instead of `"week"`